src/v/kafka/server/connection_context.cc

```diff
@@ -682,6 +684,10 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
                      ? std::make_optional<ss::sstring>(*hdr.client_id)
                      : std::nullopt,
     };
+
+    co_await ss::coroutine::switch_to(
```

Comment:
@travisdowns related to the discussion from yesterday, does this even allocate the coroutine frame if `await_ready` returns true (because we are already in the right group - ignoring for a second whether we actually are here or not)? I thought not.

Reply:
Which coroutine do you refer to? In any case there shouldn't be a coroutine frame allocated per switch_to, whether the switch happens or not. What `await_ready` returning true does is avoids an unnecessary task switch (which is probably more expensive even than a coro frame in "micro" costs, and certainly in macro costs as it breaks out of the current executing request, etc).